### ISSUE TITLE
Group inbox by repo when viewing all

### ISSUE THREAD

**Issue by gingerbeardman** [GitHawkApp/GitHawk#2411]
I find the current method of having to switch between repos in the inbox quite laborious. 

Ideally inbox would be more like the GitHub approach, where I can mark read all notifications from a repo. 

Not the best screenshot, as only one repo shown, but all I have at the moment: 

![GitHawk Upload by gingerbeardman](https://i.imgur.com/PWDkHgp.jpg)

Also see: 
#1281
#61

<details>
<summary>Bug Report Dump (Auto-generated)</summary>
<pre>
Version 1.23.0 (1541192331)
Device: iPhone 6s (iOS 12.1)
TestFlight: false
</pre>
</details>

**Comment by rnystrom** [comment]
I think this has been suggested a bunch before

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by gingerbeardman** [comment]
I found the previous two issues listed, but there could be more

**Comment by gingerbeardman** [comment]
That sounds like it would be an improvement in some ways.

But if I'm honest the the core of my request is the grouping so that I can "mark all as read" to dismiss all notifications from a repo whilst I am looking at all notifications. 

I do this so often on the GitHub website, or through BitBar ([GitHub Notifications Plugin](https://github.com/matryer/bitbar-plugins/blob/master/Dev/GitHub/notifications.30s.py)) on macOS. Not being able to do it on iOS is a bit of a pain point.

I can't find up-to-date screenshots so I'll create some of my own soon. 

Here's one from GitHub from 2012(!)

![https://s3.amazonaws.com/github-images/blog/2012/notifications-stars/inbox.png](https://s3.amazonaws.com/github-images/blog/2012/notifications-stars/inbox.png)

**Comment by gingerbeardman** [comment]
#2413 is sort of related to this?

**Comment by gingerbeardman** [comment]
I appreciate your summary. That proposition doesn't seem ant better to me as the grouping is the key thing

**Comment by j-f1** [comment]
Why not load all the pages of notifications at once?

**Comment by gingerbeardman** [comment]
Indeed. Maybe I don't use git enough, or maybe I never let my notifications build up, as I can't remember seeing the "load more" on the notifications. Just load them all (and cache them for next time)?

**Comment by gingerbeardman** [comment]
Understood, it's the switching between repos that is the friction I'd like to avoid.

Here's how a bunch of notifications looks like using the GitHub Notifications BitBar plugin (basically a shell script that can put some UI in the menu bar)

<img width="616" alt="screen shot 2018-11-11 at 10 36 31" src="https://user-images.githubusercontent.com/49612/48311931-19c98c80-e59f-11e8-8904-3b9241a8a4c7.png">

I can hold down <kbd>alt</kbd> to make any interaction "mark as read"

<img width="684" alt="untitled" src="https://user-images.githubusercontent.com/49612/48311954-67de9000-e59f-11e8-8f9b-83bc746484e5.png">

here's github.com with the same list

<img width="1040" alt="screen shot 2018-11-11 at 10 37 37" src="https://user-images.githubusercontent.com/49612/48311958-7cbb2380-e59f-11e8-86c0-bd606a0f8272.png">

and GitHawk

![img_3991](https://user-images.githubusercontent.com/49612/48311979-cd328100-e59f-11e8-9c27-4a145e5aa701.png)

Also notice how few notifications fit on screen in GitHawk! 5.5 items

**Comment by rnystrom** [comment]
If someone wants to champion design and dev of this have at it!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ijm8710** [comment]
I do think the point about how many inbox items fit on the screen at a time is a valid item to revisit

**Comment by dimirc** [comment]
One way to have full control and filter by repo, will be to load all notifications at once, but we’ll need to do multiple reqs to the notification api as the page size is limited to about 100 records per page (if I remember correctly) so in my case that I have 450 notifications, I’ll need 5 reqs to get all of them.

Checking the response sizes from api, in my example 450 notification will be about 2MB

Another way will be to parse the web version html /notifications main list, to just extract the repo names and then using the api, load the paginated results of each repo
